### PR TITLE
[CosmosDB] Update List Connection Strings output format

### PR DESCRIPTION
**Pull request by ghost**
## Description

Adds powershell changes for list connection strings response format to return new fields keyKind and type.

As part of our 2023-03-15 RP API version, we expanded the connection strings response to include two additional fields, so 4 fields total.

In other clients and SDKs we return as a 1:1 format and there is no conversion. However, in PowerShell, these connection strings were previously converted to a hash table format before being returned. Due to this we need to change the output type in order to return the two additional fields (type and kind).

Since this is a breaking change, we were unable to include it earlier - [(Discussion here)](https://github.com/Azure/azure-powershell/pull/21990/commits/ff3ef0afb546d33303f76e35acc97c5a8443f360#r1230642059).

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by msJinLei** [comment]
@carjackson-msft do you sent the breaking change request?

**Comment by msJinLei** [comment]
Refer to https://github.com/Azure/azure-powershell/pull/24108

File: src/CosmosDB/CosmosDB/CosmosDBAccount/GetAzCosmosDBAccountKey.cs [hunk 1]
```diff
@@ -24,8 +25,7 @@
 
 namespace Microsoft.Azure.Commands.CosmosDB
 {
-    [GenericBreakingChangeWithVersionAttribute("Output type for -Type ConnectionStrings will be changed to List<DatabaseAccountConnectionString> in next major version.", Constants.DeprecateByAzVersion12, Constants.DeprecateByVersion2)]
-    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBAccountKey", DefaultParameterSetName = NameParameterSet), OutputType(typeof(Hashtable))]
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBAccountKey", DefaultParameterSetName = NameParameterSet), OutputType(typeof(Hashtable), typeof(List<DatabaseAccountConnectionString>))]
```
Comment (msJinLei) [inline src/CosmosDB/CosmosDB/CosmosDBAccount/GetAzCosmosDBAccountKey.cs:28]: Please submit a separate PR for the breaking change notification only.

Comment (msJinLei) [inline src/CosmosDB/CosmosDB/CosmosDBAccount/GetAzCosmosDBAccountKey.cs:28]: Remove the breaking change notification for this PR.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by msJinLei** [comment]
> > Refer to #24108
> 
> I've updated that PR with the proper breaking change notification formatting. How do I get this PR included in the next breaking change release? @msJinLei

current PR is going to be merged in the next major release in May

File: src/CosmosDB/CosmosDB/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,7 @@
 -->
 
 ## Upcoming Release
+* [Breaking Change] Output type for -Type ConnectionStrings is changed to List<DatabaseAccountConnectionString> to expose keyKind and type fields
```
Comment (YanaXu) [inline src/CosmosDB/CosmosDB/ChangeLog.md:22]: Hi @carjackson-msft, it's not a breaking now aftrer your update. Could you update the changlog line to remove "Breaking Change"?

Comment (YanaXu) [reply src/CosmosDB/CosmosDB/ChangeLog.md:22]: And please update the line to more user-friendly.

**Comment by YanaXu** [comment]
Hi @carjackson-msft, please update it according to the comment and resolve the conflict.

**Pull request closed without merging**